src/bin/lockin-external.rs

```diff
@@ -4,31 +4,106 @@
 
 use stm32h7xx_hal as hal;
 
+use miniconf::{
+    embedded_nal::{IpAddr, Ipv4Addr},
+    minimq, MqttInterface, StringSet,
+};
+
+use serde::Deserialize;
 use stabilizer::{hardware, hardware::design_parameters};
 
 use dsp::{lockin::Lockin, rpll::RPLL, Accu};
+
 use hardware::{
-    Adc0Input, Adc1Input, Dac0Output, Dac1Output, InputStamper, AFE0, AFE1,
+    Adc0Input, Adc1Input, AfeGain, Dac0Output, Dac1Output, InputStamper, AFE0,
+    AFE1,
 };
 
+#[derive(Debug, Clone, Copy, Deserialize, StringSet)]
+pub struct DspData {
+    // frequency settling time (log2 counter cycles)
+    frequency_settling_time: u8,
+
+    // phase settling time
+    phase_settling_time: u8,
+
+    // Harmonic index of the LO: -1 to _de_modulate the fundamental (complex conjugate)
+    harmonic: i32,
+
+    // Demodulation LO phase offset
+    phase_offset: i32,
+
+    // Log2 lowpass time constant
+    time_constant: u8,
+}
```

Comment:
We don't necessarily need to bother with the other binaries in this PR yet. Having just `dual-iir` for now would be ok.

Reply:
I was thinking the same thing - otherwise, things start to get a big big. Let me yank all of the lockin-external changes to a different branch so we can keep things tidy.